Update tmuxPlugins, en masse.

judemille opened the pull request:
## Description of changes

Everything in tmuxPlugins, that has a new revision, has been updated.
The group seemed neglected, so I gave it some love.

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

judemille committed:
tmuxPlugins.battery: unstable-2019-07-04 -> unstable-2023-12-01

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.continuum: unstable-2022-01-25 -> unstable-2024-01-20

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.better-mouse-mode: unstable-2021-08-02 -> unstable-2017-10-16

Version was previously unstable-2021-08-02, despite referencing a commit
made 2017-10-16. That commit is still latest, so the version has been
updated to unstable-2017-10-16.

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.copycat: unstable-2020-01-09 -> unstable-2020-07-24

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.dracula: 2.2.0 -> 2.3.0

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.extrakto: unstable-2021-04-04 -> unstable-2024-05-09

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.fingers: 1.0.1 -> 2.1.5

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.fpp: unstable-2016-03-08 -> unstable-2024-01-29

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.fuzzback: unstable-2022-11-21 -> unstable-2024-01-20

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.fzf-tmux-url: unstable-2021-12-27 -> unstable-2024-04-14

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.jump: 2020-06-26 -> unstable-2023-05-09

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.logging: unstable-2019-04-19 -> 2.1.0-unstable-2021-06-10

judemille committed:
tmuxPlugins.mode-indicator: unstable-2021-10-01 -> unstable-2023-03-24

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.maildir-counter: unstable-2016-11-25 -> unstable-2021-03-15

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.online-status: unstable-2018-11-30 -> unstable-2018-11-30

Somehow, there was another commit, on the same day as the currently
posted version, that was not tracked in Nixpkgs, in the
tmux-online-status repo. It feels very weird to bump the rev without
changing the version, but I don't know of any derivation revision number
to bump, so I guess this is how it goes.

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.open: unstable-2019-12-02 -> 3.0.0-unstable-2022-08-22

judemille committed:
tmuxPlugins.onedark-theme: unstable-2020-06-07 -> unstable-2020-02-25

Same thing as with better-mouse-mode. The commit is from the date I'm
updating to. Whoever put this derivation in, did not properly version
it.

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.pain-control: unstable-2020-02-18 -> unstable-2021-08-09

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.power-theme: unstable-2020-11-18 -> unstable-2024-04-14

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.prefix-highlight: unstable-2021-03-30 -> unstable-2024-01-14

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.resurrect: unstable-2022-05-01 -> unstable-2023-03-06

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.sensible: unstable-2017-09-05 -> unstable-2022-08-14

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.sessionist: unstable-2017-12-03 -> 2.3.0-unstable-2023-05-02

judemille committed:
tmuxPlugins.sidebar: unstable-2018-11-30 -> 0.7.0-unstable-2022-12-08

judemille committed:
tmuxPlugins.tmux-colors-solarized: unstable-2019-07-14 -> unstable-2022-06-10

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.tmux-fzf: unstable-2023-10-24 -> unstable-2024-04-15

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins.vim-tmux-focus-events: unstable-2020-10-05 -> 1.0.0-unstable-2021-04-27

judemille committed:
tmuxPlugins.vim-tmux-navigator: unstable-2022-08-01 -> 1.0-unstable-2024-04-13

judemille committed:
tmuxPlugins.yank: unstable-2021-06-20 -> 2.3.0-unstable-2023-07-19

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille committed:
tmuxPlugins: remove mkDerivation

Previously a deprecation `throw`, but that causes issues for mass
building. It's not used anywhere in here, so I just got rid of it.

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille commented:
Note: the `tmuxPlugins.mkDerivation` deprecation attribute was removed, since it prevented building the entire attribute set, and nothing inside `tmuxPlugins` uses it anymore.

judemille commented:
@ofborg build tmuxPlugins

Realizing I can do this. Hopefully not considered being non-"gentle" with the thing. These derivations *should* be small. Apologies if I'm not meant to do that.

judemille commented:
It would appear that, in my rewriting of Git history to make *my* commits match the guidelines, I've accidentally pulled in some old commits and signed them. No idea how to fix that.

judemille commented:
Okay, fixed that strange rebase issue. Apologies for the cluttered PR log, but now the Git history is sensible.

h7x4 reviewed (commented):
This looks good!

Nothing important, but it would be nice if the updated packages could get an SRI hash.

There's also a somewhat new convention that unstable packages that are part of an earlier version tag should be versioned so that it mentions the latest stable release. Have a look at https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming. I've gone through and marked the ones I found.

Some of the commit messages are also a bit inconsistent, where most versions are labeled with `unstable-` prefixes, but others not.

judemille commented on pkgs/misc/tmux-plugins/default.nix:
```diff
@@ -418,26 +416,26 @@ in rec {
     pluginName = "onedark-theme";
     rtpFilePath = "tmux-onedark-theme.tmux";
-    version = "unstable-2020-06-07";
+    version = "unstable-2020-02-25";
```
The original version was erroneous -- the referenced, and still latest commit, is from 2020-02-25.

judemille commented on pkgs/misc/tmux-plugins/default.nix:
```diff
@@ -50,22 +50,20 @@ let
   better-mouse-mode = mkTmuxPlugin {
     pluginName = "better-mouse-mode";
-    version = "unstable-2021-08-02";
+    version = "unstable-2017-10-16";
```
As with onedark-theme, the referenced (and latest) commit is from 2017-10-16.

judemille commented on pkgs/misc/tmux-plugins/default.nix:
```diff
@@ -418,26 +416,26 @@ in rec {
     src = fetchFromGitHub {
       owner = "tmux-plugins";
       repo = "tmux-online-status";
-      rev = "ea86704ced8a20f4a431116aa43f57edcf5a6312";
```
Looks like it did indeed have an invalid version before. I'll change the version to follow that convention.

judemille commented:
@h7x4 
> Nothing important, but it would be nice if the updated packages could get an SRI hash.

I'll see if I can do that in a way that doesn't require me to update by hand.

> There's also a somewhat new convention that unstable packages that are part of an earlier version tag should be versioned so that it mentions the latest stable release. Have a look at https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-naming. I've gone through and marked the ones I found.

I'll incorporate those changes.

> Some of the commit messages are also a bit inconsistent, where most versions are labeled with `unstable-` prefixes, but others not.

With the current version of the Git history (after my goofy rebasing), the versions mentioned in the commit messages should be accurate to the versions inside the Nix files.

h7x4 commented:
@judemille Do you mind if I push some changes to the branch?

judemille commented:
@h7x4 Go ahead.

Ellipse0934 commented:
My PR (#290575) could possibly go in here or could be independently merged first. Probably someone here could review that one.

NovaViper commented:
Hey @judemille I think the PR needs to be rebased against master, since #290575 merged a few hours ago

marcusramberg commented:
Would be awesome to get the tmuxPlugins in a bit better shape :)

judemille commented:
I'll rebase when I get a chance. Suggestions on updating the hash methods en masse would be appreciated.

marcusramberg commented:
> I'll rebase when I get a chance. Suggestions on updating the hashes to SRI en masse would be appreciated.

Not sure what editor you use, but in nvim this sorts it 

```
:%s/sha256 = "/hash = "sha256-/
```

judemille commented:
Conflicts fixed, some new updates added, rebased, and SRIs for all changes. Coming soon to a PR near you!

judemille committed:
tmuxPlugins.urlview: remove

urlview was removed from nixpkgs by facb7a38b3a0a7f276f8a7f7902cb126b51e87f4.
the tmux plugin cannot exist without it.

Signed-off-by: Julia DeMille <me@jdemille.com>

judemille commented:
Note: the plugin `urlview` had to be removed, since `urlview` was removed from Nixpkgs. Unsure if the procedure I followed is best.

marcusramberg commented:
Seems this pr now reverts #290575, otherwise looks good to me.

```
10 packages built:
tmuxPlugins.better-mouse-mode tmuxPlugins.catppuccin tmuxPlugins.continuum tmuxPlugins.extrakto tmuxPlugins.fuzzback tmuxPlugins.jump tmuxPlugins.mode-indicator tmuxPlugins.resurrect tmuxPlugins.tmux-fzf tmuxPlugins.vim-tmux-focus-events
```

judemille commented:
> Seems this pr now reverts #290575, otherwise looks good to me.

I'll double check that and rebase again.

jnsgruk commented:
@judemille I think the same from this PR: https://github.com/NixOS/nixpkgs/pull/312197 - looks like this would roll the version back.

frankitox commented:
Hi! I don't want to bring too much noise to the PR. But I just tried to update tmux-fingers based on the code found here and it didn't work, basically this:

```nix
mkTmuxPlugin rec {
    pluginName = "fingers";
    rtpFilePath = "tmux-fingers.tmux";
    version = "2.1.5";
    src = fetchFromGitHub {
      owner = "Morantron";
      repo = "tmux-fingers";
      rev = version;
      hash = "sha256-gR3u5IVgFxd6uj7l8Ou8GnEvh8AkjRFgIWKCviISweQ=";
      fetchSubmodules = true;
    };
    nativeBuildInputs = [ pkgs.makeWrapper ];
    postInstall = ''
      chmod +x $target/tmux-fingers.tmux
      wrapProgram $target/tmux-fingers.tmux \
        --prefix PATH : ${with pkgs; lib.makeBinPath (
          [ gawk ] ++ lib.optionals stdenv.isDarwin [ reattach-to-user-namespace ]
        )}
    '';
};
```

I had to use the old code and only change the version string and it worked properly:

```nix
pkgs.tmuxPlugins.mkTmuxPlugin rec {
  pluginName = "tmux-fingers";
  rtpFilePath = "load-config.tmux";
  version = "2.1.5";
  src = pkgs.fetchFromGitHub {
    owner = "Morantron";
    repo = "tmux-fingers";
    rev = "${version}";
    sha256 = "sha256-1YMh6m8M6FKf8RPXsOfWCVC5CXSr/MynguwkG7O+oEY=";
  };
  nativeBuildInputs = [ pkgs.makeWrapper pkgs.crystal pkgs.shards ];
  postInstall = ''
    shards build --production
    rm -rf $target/* $target/.*
    cp -r bin $target/bin
    echo "$target/bin/${pluginName} load-config" > $target/${rtpFilePath}
    chmod +x $target/${rtpFilePath}
    wrapProgram $target/${rtpFilePath} \
      --prefix PATH : ${with pkgs; lib.makeBinPath (
        [ gawk ] ++ lib.optionals stdenv.isDarwin [ reattach-to-user-namespace ]
      )}
  '';
}
```
